core/lib/src/cookies.rs

```diff
@@ -163,6 +164,7 @@ impl<'a> Clone for CookieJar<'a> {
             jar: self.jar.clone(),
             ops: Mutex::new(self.ops.lock().clone()),
             config: self.config,
+            proxy_secure: self.proxy_secure,
```

Comment:
Why do we need this?

Reply:
I wholeheartedly agree that this isn't the nicest solution, but when working on it I couldn't come up with a better one. The problem I wanted to solve is that only the `Request` knows whether we are in a secure context (as forwarded by the proxy) but the `CookieJar` itself has no way to determine this.

I would love to remove this, and if you could provide some guidance in a better way to determine whether we are in a secure context (when setting the defaults) I will make changes to do so. Or feel free to do them yourself

Reply:
Superseded by https://github.com/rwf2/Rocket/pull/2519/commits/678b7c1f8dfe59d43a6295c0221d102ba720d520#diff-c8751afca2b03eeea5b65a1e8cc20f0cf291dcce2a129b96d3d405247c1d7222R178